PokeDemoTests/PokemonListPresenterTests.swift

```diff
@@ -139,20 +139,120 @@ final class PokemonListPresenterTests: XCTestCase {
+        // Given
+        let mockDelegate = MockPokemonListViewPresenterDelegate()
+        let imageData = try? Data(contentsOf: URL(fileURLWithPath: Bundle(for: type(of: self)).path(forResource: "test", ofType: "png") ?? ""))
+        presenter = PokemonListPresenter(service: mockService)
+        presenter?.delegate = mockDelegate
+        presenter?.didSetupCellWith(pokemon: Pokemon(name: "abc", url: "testURL"), completion: { data in
+            XCTAssertEqual(data, imageData)
+        })
+        
+        // When service returns successful response
+        let jsonData = JSONString.successWithPokemonInfo.data(using: .utf8)
+        mockService.onCompletion?(jsonData, nil)
+        
+        // Then
+        XCTAssertEqual(mockService.sendRequestWithJSONIsCalledCount, 2)
+        
+        mockService.onCompletion?(imageData, nil)
+    }
+    
+    func testGetPokemonImageReturnsError(){
+        // Given
+        let mockDelegate = MockPokemonListViewPresenterDelegate()
+        presenter = PokemonListPresenter(service: mockService)
+        presenter?.delegate = mockDelegate
+        presenter?.didSetupCellWith(pokemon: Pokemon(name: "abc", url: "testURL"), completion: { data in
+        })
+        
+        // When service returns error
+        mockService.onCompletion?(nil, TestError())
+        
+        // Then
+        XCTAssertEqual(mockService.sendRequestWithJSONIsCalledCount, 1)
+    }
+    
+    func testGetPokemonImageReturnsNil(){
+        // Given
+        let mockDelegate = MockPokemonListViewPresenterDelegate()
+        presenter = PokemonListPresenter(service: mockService)
+        presenter?.delegate = mockDelegate
+        presenter?.didSetupCellWith(pokemon: Pokemon(name: "abc", url: "testURL"), completion: { data in
+        })
+        
+        // When service returns nil
+        mockService.onCompletion?(nil, nil)
+        
+        // Then
+        XCTAssertEqual(mockService.sendRequestWithJSONIsCalledCount, 1)
+    }
+    
+    func testGetPokemonImageReturnsResponseWithIncorrectJSON(){
+        // Given
+        let mockDelegate = MockPokemonListViewPresenterDelegate()
+        presenter = PokemonListPresenter(service: mockService)
+        presenter?.delegate = mockDelegate
+        presenter?.didSetupCellWith(pokemon: Pokemon(name: "abc", url: "testURL"), completion: { data in
+        })
+        
+        // When service returns incorrect response
+        let jsonData = JSONString.errorWithIncorrectPokemonInfo.data(using: .utf8)
+        mockService.onCompletion?(jsonData, nil)
+        
+        // Then
+        XCTAssertEqual(mockService.sendRequestWithJSONIsCalledCount, 1)
+    }
+    
+    func testGetPokemonImageReturnsIncorrectResponseForImage(){
+        // Given
+        let mockDelegate = MockPokemonListViewPresenterDelegate()
+        let imageData = try? Data(contentsOf: URL(fileURLWithPath: Bundle(for: type(of: self)).path(forResource: "testing", ofType: "png") ?? ""))
```

Comment:
Move it in stub. The way you move in success and failure response

Reply:
Covered in 1127aac0a5568779fa523763eb1c01779fb18fb5